### ISSUE TITLE
linter: Issue parsing HTML custom tag that starts with `script`

### ISSUE THREAD

**Issue by Antti-Palola** [oxc-project/oxc#9837]
### What version of Oxlint are you using?

0.16.0

### What command did you run?

`npx oxlint@latest`

### What does your `.oxlintrc.json` config file look like?

Config doesn't matter

### What happened?

I have a couple of Vue components prefixed with "Script". An unexpected token error is thrown when parsing a file like that (line 2)
```
<template>
  <script-view />
</template>
<script lang="ts">
import ScriptView from "./ScriptView.vue"
export default defineComponent({
  components: { ScriptView }
});
```
In another component a component called `<scripts-list />` is used with the same error.
